src/auth/auth.service.ts

```diff
@@ -1,4 +1,43 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, BadRequestException, InternalServerErrorException } from '@nestjs/common';
+import { UserRepository } from './user.repository';
+import { InjectRepository } from '@nestjs/typeorm';
+import { UserRegisterDto } from './dto/user-register.dto';
+import * as authError from '../constants/auth.constants';
+import HashHelper from '../helpers/hash.helper';
+import TokenHelper from '../helpers/token.helper';
 
 @Injectable()
-export class AuthService {}
+export class AuthService {
+    constructor(
+        @InjectRepository(UserRepository)
+        private userRepository: UserRepository
+    ){}
+    async registerUser(userRegisterDto: UserRegisterDto){
+       try {
+        const {firstname, lastname, username, email, password} = userRegisterDto;
+
+        const user = await this.userRepository.getUser(email);
+
+        if(user) {
+            throw new BadRequestException(authError.USER_EXISTS);
+        }
+
+        const userData: UserRegisterDto = { firstname, lastname, username,email, password: HashHelper.hashPassword(password), verified: false};
+        const newUser = await this.userRepository.createUser(userData);
+
+        const payload = {
+            user: { id: newUser.id, username: newUser.username, status: newUser.verified },
```

Comment:
the status key used here is a bit ambiguous. I think you should just make it the same as verified